### PR TITLE
Homework jv-storage done

### PR DESCRIPTION
**Pull request by Oleh8**

### CODE REVIEW

File: src/main/java/core/basesyntax/impl/Pair.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
+package core.basesyntax.impl;
+
+public class Pair<K, V> {
```
Comment (aaananas) [inline src/main/java/core/basesyntax/impl/Pair.java:3]: lets make this class an inner class in StorageImpl 
you wont need getters that way

File: src/main/java/core/basesyntax/impl/StorageImpl.java [hunk 2]
```diff
@@ -3,17 +3,43 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private final int maxSize = 10;
```
Comment (aaananas) [inline src/main/java/core/basesyntax/impl/StorageImpl.java:6]:
```suggestion
    private static final int MAX_SIZE = 10;
```
this should be a constant

File: src/main/java/core/basesyntax/impl/StorageImpl.java [hunk 3]
```diff
@@ -3,17 +3,43 @@
...
+    private Pair[] pairArr;
```
Comment (aaananas) [inline src/main/java/core/basesyntax/impl/StorageImpl.java:7]:
```suggestion
    private Pair<K, V>[] pairArray;
```
use full names

File: src/main/java/core/basesyntax/impl/StorageImpl.java [hunk 4]
```diff
@@ -3,17 +3,43 @@
...
+                    || (pairArr[i].getKey() != null && pairArr[i].getKey().equals(key)))) {
+                pairArr[i] = new Pair<>(key, value);
+                return;
+            } else if (pairArr[i] == null) {
+                pairArr[i] = new Pair<>(key, value);
+                actualSize++;
+                return;
+            }
+        }
     }
```
Comment (aaananas) [inline src/main/java/core/basesyntax/impl/StorageImpl.java:28]:
```suggestion
    public void put(K key, V value) {
        for (int i = 0; i < actualSize; i++) {
            if (pairArr[i].getKey() == key)
                    || (pairArr[i].getKey() != null && pairArr[i].getKey().equals(key)))) {
                pairArr[i] = new Pair<>(key, value);
                return;
            } 
        }
        pairArr[i] = new Pair<>(key, value);
        actualSize++;
    }
```

**Review by aaananas — CHANGES_REQUESTED** [review]

File: src/main/java/core/basesyntax/impl/StorageImpl.java (hunk 2 above)
Comment (Oleh8) [reply src/main/java/core/basesyntax/impl/StorageImpl.java:6]: a cant pass tests if I put here capital letters

File: src/main/java/core/basesyntax/impl/StorageImpl.java (hunk 4 above)
Comment (Oleh8) [reply src/main/java/core/basesyntax/impl/StorageImpl.java:28]: this:
 pairArr[i] = new Pair<>(key, value);
is out of for loop...  does not work that way

**Pull request closed without merging**